akka-contrib/src/test/scala/akka/contrib/circuitbreaker/sample/CircuitBreakerAskWithFailure.scala

```diff
@@ -0,0 +1,38 @@
+package akka.contrib.circuitbreaker.sample
+
+import akka.actor.{ Actor, ActorLogging, ActorRef }
+import akka.contrib.circuitbreaker.CircuitBreakerActor._
+import akka.util.Timeout
+
+import scala.concurrent.duration._
+import scala.util.{ Failure, Success }
+
+//#ask-with-failure-sample
+class CircuitBreakerAskWithFailure(potentiallyFailingService: ActorRef) extends Actor with ActorLogging {
+  import SimpleService._
+  import akka.pattern._
+
+  implicit val askTimeout: Timeout = 2.seconds
+
+  val serviceCircuitBreaker =
+    context.actorOf(
+      CircuitBreakerActorBuilder(maxFailures = 3, callTimeout = askTimeout, resetTimeout = 30.seconds).propsForTarget(potentiallyFailingService),
+      "serviceCircuitBreaker")
+
+  import context.dispatcher
+
+  override def receive: Receive = {
+    case AskFor(requestToForward) ⇒
+      (serviceCircuitBreaker ? Request(requestToForward)).failForOpenCircuit.mapTo[String].onComplete {
```

Comment:
I'm not completely convienved about `failForOpenCircuit` yet (adding methods to Future via an implicit via importing `import akka.contrib.circuitbreaker.CircuitBreakerActor._` seems hard to discover). I see why it's useful - to keep the fluent-api going to the mapTo etc, how we obtain this method is hard to discover